tests/sentry/api/endpoints/test_project_stacktrace_link.py

```diff
@@ -213,3 +214,163 @@ def test_file_no_stack_root_match(self, mock_integration):
         assert not response.data["sourceUrl"]
         assert response.data["error"] == "stack_root_mismatch"
         assert response.data["integrations"] == [serialized_integration(self.integration)]
+
+
+@region_silo_test
+class ProjectStacktraceLinkTestMobile(APITestCase):
+    endpoint = "sentry-api-0-project-stacktrace-link"
+
+    def setUp(self):
+        self.integration = Integration.objects.create(provider="example", name="Example")
+        self.integration.add_organization(self.organization, self.user)
+        self.oi = OrganizationIntegration.objects.get(integration_id=self.integration.id)
+
+        self.repo = self.create_repo(
+            project=self.project,
+            name="getsentry/sentry",
+        )
+        self.repo.integration_id = self.integration.id
+        self.repo.provider = "example"
+        self.repo.save()
+
+        self.code_mapping1 = self.create_code_mapping(
+            organization_integration=self.oi,
+            project=self.project,
+            repo=self.repo,
+            stack_root="",
+            source_root="",
+        )
+
+        self.login_as(self.user)
+
+    def expected_configurations(self, code_mapping) -> Mapping[str, Any]:
+        return {
+            "defaultBranch": "master",
+            "id": str(code_mapping.id),
+            "integrationId": str(self.integration.id),
+            "projectId": str(self.project.id),
+            "projectSlug": self.project.slug,
+            "provider": serialized_provider(),
+            "repoId": str(self.repo.id),
+            "repoName": self.repo.name,
+            "sourceRoot": code_mapping.source_root,
+            "stackRoot": code_mapping.stack_root,
+        }
+
+    def test_file_not_found_munge_frame_fallback_success_android(self):
+        ctx = {
+            "file": "file.java",
+            "filename": "file.java",
+            "commit_id": None,
+            "platform": "java",
+            "sdk_name": None,
+            "abs_path": None,
+            "module": "usr.src.getsentry.file",
+            "package": None,
+        }
+
+        munged_url = try_path_munging(self.code_mapping1, "file.java", ctx)
+
+        with mock.patch.object(
+            ExampleIntegration,
+            "get_stacktrace_link",
+            return_value="https://example.com/getsentry/sentry/blob/master/usr/src/getsentry/file.java",
```

Comment:
Mock the return value from `get_stacktrace_link` so the rest of the process can be tested. Without this, `get_stacktrace_link` always return `None` because of [this](https://github.com/getsentry/sentry/blob/c294f8f3ee28fa2ee375e0808e7dc48120a7ab19/src/sentry/integrations/example/integration.py#L161)

Reply:
This is correct.